src/app/shared/code-editor-list/code-editor-list.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Input } from '@angular/core';
+import { SourceFile } from '../../shared/swagger/model/sourceFile';
+
+@Component({
+  selector: 'app-code-editor-list',
+  templateUrl: './code-editor-list.component.html',
+  styleUrls: ['./code-editor-list.component.scss']
+})
+export class CodeEditorListComponent {
+  @Input() sourcefiles: Array<any>;
+  @Input() editing: boolean;
+  @Input() descriptorType: string;
+  @Input() fileType: string;
+  constructor() { }
+
+  /**
+   * Adds a new file editor
+   */
+  addFile() {
+    let newFilePath = this.getDefaultPath();
+    if (this.sourcefiles.length === 0 && this.fileType === 'descriptor') {
+      newFilePath = '/Dockstore' + newFilePath;
+    }
+    const newSourceFile = {
+      content: '',
+      path: newFilePath,
+      type: this.getFileType()
+    };
+
+    if (this.sourcefiles === undefined) {
+      this.sourcefiles = [];
+    }
+    this.sourcefiles.push(newSourceFile);
+  }
+
+  /**
+   * Deletes the file at the given index
+   * @param  index index of file to delete
+   */
+  deleteFile(index: number) {
+    this.sourcefiles[index].content = null;
+  }
+
+  /**
+   * Get the file type enum
+   * @return {string} The file type enum
+   */
+  getFileType() {
+    if (this.fileType === 'descriptor') {
+      return 'DOCKSTORE_' + this.descriptorType.toUpperCase();
```

Comment:
Please make an issue to make this kind of thing case insensitive in the webservice.